petastorm/arrow_reader_worker.py

```diff
@@ -173,6 +173,10 @@ def process(self, piece_index, worker_predicate, shuffle_row_drop_partition):
     def _check_shape_and_ravel(x, field):
         if not isinstance(x, np.ndarray):
             raise ValueError('field {name} must be numpy array type.'.format(name=field.name))
+        if None in field.shape:
+            raise ValueError(f'All dimensions of a shape: {field.shape} in: {field.name} field must be constant. '
+                             f'If a dimension is variable, we won\'t be able to coalesce rows when preparing a batch.')
+
         if x.shape != field.shape:
```

Comment:
Should we use `x.shape != tuple(field.shape)`, in case of field.shape may be a list ?